### PR TITLE
Fixed missing special resources crashing the server

### PR DESCRIPTION
**Pull request by JohannesGaessler**
Fixes https://github.com/DistributedTaskScheduling/JobAdder/issues/184.

AddCommand.execute now checks if a job can be executed at all.
If the requested special resources don't exist at all or in insufficient quantity the job is not added.

### CODE REVIEW

File: src/ja/user/message/add.py [hunk 1]
```diff
@@ -43,6 +44,21 @@ def execute(self, database: ServerDatabase) -> Response:
         if db_job_id is not None:
             return Response(result_string="Job with id %s already exists" % job.uid,
                             is_success=False)
+
+        if job.scheduling_constraints.special_resources is not None:
```
Comment (ammen99) [inline src/ja/user/message/add.py]: When is this None? In tests?

Comment (JohannesGaessler) [reply src/ja/user/message/add.py]: It seems I misremembered whether special_resources is allowed to be None.
I re-ran unittests without this line and everything was working as intended.

**Comment by JohannesGaessler** [comment]
Somewhere in the dispatcher a dictionary entry was retrieved for special resources.
If the special resources didn't exist this caused a KeyError.

**Comment by ammen99** [comment]
> Somewhere in the dispatcher a dictionary entry was retrieved for special resources.
> If the special resources didn't exist this caused a KeyError.

Dispatcher???

**Comment by JohannesGaessler** [comment]
Sorry, I meant Scheduler.
This is the full stack trace:
Exception in thread server-main:
Traceback (most recent call last):
`  File "/usr/lib/python3.8/threading.py", line 932, in _bootstrap_inner
    self.run()
  File "/usr/lib/python3.8/threading.py", line 870, in run
    self._target(*self._args, **self._kwargs)
  File "/home/johannesg/Documents/Informatik/pse/JobAdder/src/ja/server/main.py", line 99, in run
    self._handler.main_loop()
  File "/home/johannesg/Documents/Informatik/pse/JobAdder/src/ja/common/proxy/command_handler.py", line 90, in main_loop
    response_dict = self._check_exit_or_process_command(
  File "/home/johannesg/Documents/Informatik/pse/JobAdder/src/ja/common/proxy/command_handler.py", line 64, in _check_exit_or_process_command
    return self._process_command_dict(command_dict, type_name, username)
  File "/home/johannesg/Documents/Informatik/pse/JobAdder/src/ja/server/proxy/command_handler.py", line 89, in _process_command_dict
    response = self._process_user_message(command_dict, type_name, username)
  File "/home/johannesg/Documents/Informatik/pse/JobAdder/src/ja/server/proxy/command_handler.py", line 81, in _process_user_message
    return self._execute_command(user_command)
  File "/home/johannesg/Documents/Informatik/pse/JobAdder/src/ja/server/proxy/command_handler.py", line 58, in _execute_command
    r_dict = command.execute(self._database)
  File "/home/johannesg/Documents/Informatik/pse/JobAdder/src/ja/user/message/add.py", line 63, in execute
    job.uid = database.update_job(job)
  File "/home/johannesg/Documents/Informatik/pse/JobAdder/src/ja/server/database/sql/database.py", line 258, in update_job
    self._call_scheduler()
  File "/home/johannesg/Documents/Informatik/pse/JobAdder/src/ja/server/database/sql/database.py", line 339, in _call_scheduler
    self.scheduler_callback(self)
  File "/home/johannesg/Documents/Informatik/pse/JobAdder/src/ja/server/scheduler/scheduler.py", line 85, in reschedule
    new_schedule = self._algorithm.reschedule_jobs(runnable_entries, available_machines, self.special_resources)
  File "/home/johannesg/Documents/Informatik/pse/JobAdder/src/ja/server/scheduler/default_algorithm.py", line 153, in reschedule_jobs
    next_special_resources[resource] -= 1
KeyError: 'license'`

**Pull request merged**